File: 2024-10-October-GFG-POTD/October 25 - Alternative Sorting/Solution.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;

class Main {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());

        while (t-- > 0) {

            ArrayList<Integer> array1 = new ArrayList<Integer>();
            String line = read.readLine();
            String[] tokens = line.split(" ");
            for (String token : tokens) {
                array1.add(Integer.parseInt(token));
            }
            ArrayList<Integer> v = new ArrayList<Integer>();
            int[] arr = new int[array1.size()];
            int idx = 0;
            for (int i : array1) arr[idx++] = i;

            v = new Solution().alternateSort(arr);

            for (int i = 0; i < v.size(); i++) System.out.print(v.get(i) + " ");

            System.out.println();
            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public static ArrayList<Integer> alternateSort(int[] arr) {

        // Your code goes here
         Arrays.sort(arr);
        
        ArrayList<Integer> result = new ArrayList<>();
        
        int i = 0; // Pointer to the smallest element
        int j = arr.length - 1; // Pointer to the largest element
        
        while (i <= j) {
            // Append the largest element
            if (i != j) {
                result.add(arr[j--]);
            }
            
            // Append the smallest element
            result.add(arr[i++]);
        }
        
        return result;
    }
}

File: 2024-10-October-GFG-POTD/October 25 - Alternative Sorting/Solution.py
# Driver Code Starts
# Initial Template for Python

t = int(input())
for _ in range(t):
    array1 = list(map(int, input().split()))
    
    v = Solution().alternateSort(array1)
    
    print(" ".join(map(str, v)))
    print("~")

# } Driver Code Ends

# User function Template for Python

class Solution:
    def alternateSort(self, arr):
        # Sort the array
        arr.sort()

        result = []
        i = 0  # Pointer to the smallest element
        j = len(arr) - 1  # Pointer to the largest element

        while i <= j:
            # Append the largest element
            if i != j:
                result.append(arr[j])
                j -= 1

            # Append the smallest element
            result.append(arr[i])
            i += 1

        return result
